src/main/java/bg/nbu/gradebook/domain/entities/School.java

```diff
@@ -0,0 +1,37 @@
+package bg.nbu.gradebook.domain.entities;
+
+import static javax.persistence.CascadeType.ALL;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.JoinColumn;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+@Entity
+@Getter
+@Setter
+@NoArgsConstructor
+@Table(name = "schools")
+public class School extends BaseEntity {
+
+    @Column(name = "name", unique = true, nullable = false)
+    private String name;
+
+    @Column(name = "address", unique = true, nullable = false)
+    private String address;
+
+    @Column(name = "director", unique = true, nullable = false)
+    private User director;
+
+    @OneToMany(cascade = ALL, orphanRemoval = true)
+    @JoinColumn(name = "class_id")
```

Comment:
Shouldn't it be `school_id` as the classes will have a foreign key in their table? 
Also you can skip the `@JoinColumn` if you like and Spring will generate the correct name.